filament/backend/src/vulkan/VulkanBuffer.h

```diff
@@ -29,13 +29,33 @@ class VulkanBuffer {
     VulkanBuffer(VulkanContext& context, VulkanStagePool& stagePool, VkBufferUsageFlags usage,
             uint32_t numBytes);
     ~VulkanBuffer();
-    void loadFromCpu(const void* cpuData, uint32_t byteOffset, uint32_t numBytes);
+    void terminate(VulkanContext& context);
+    void loadFromCpu(VulkanContext& context, VulkanStagePool& stagePool,
+            const void* cpuData, uint32_t byteOffset, uint32_t numBytes) const;
     VkBuffer getGpuBuffer() const { return mGpuBuffer; }
+
 private:
-    VulkanContext& mContext;
-    VulkanStagePool& mStagePool;
+    constexpr VkAccessFlags getAccessFlags() const noexcept {
```

Comment:
These two methods are vaguely named, there are several access flags (dst & src) and it's unclear that these are being used for a pipeline barrier.

I'd rather see a high-level "if" with vkCmdPipelineBarrier in each branch, so that you can see all the arguments in one place. Generally speaking, pipeline barriers are very hard to read, and IMO it's fine to be verbose and repetitive if it helps readability.

I also want to take care to preserve the comment I wrote in VulkanUniformBuffer::loadFromCpu because future code readers will wonder why we're using VK_PIPELINE_STAGE_ALL_GRAPHICS_BIT (it's for Mali).